src/service/resource.js

```diff
@@ -695,10 +695,10 @@ export class Resource {
     const viewportRatio = this.getDistanceViewportRatio();
     const keys = Object.keys(this.withViewportDeferreds_);
     for (let i = 0; i < keys.length; i++) {
-      const viewportInt = dev().assertNumber(parseInt(keys[i], 10));
-      if (this.isWithinViewportRatio(viewportInt, viewportRatio)) {
-        this.withViewportDeferreds_[viewportInt].resolve();
-        delete this.withViewportDeferreds_[viewportInt];
+      const viewportNum = dev().assertNumber(parseFloat(keys[i], 10));
```

Comment:
parseFloat(1.25, 10) = 1.25, which leads to `this.withViewportDeferreds_[1.25]`. I think we should keep using `parseInt`

Reply:
nvm, just learnt that the key here is a string, where we are expecting this.withViewportDeferreds_['1.25']